Doc/c-api/unicode.rst

```diff
@@ -607,6 +607,20 @@ APIs:
    decref'ing the returned objects.
 
 
+.. c:function:: void PyUnicode_Append(PyObject **p_left, PyObject *right)
+
+   Append the string *right* to the end of *p_left*.
+   *p_left* must point to a :term:`strong reference` to a Unicode object.
+
+   On error, set *\*p_left* to ``NULL`` and set an exception.
```

Comment:
IMO, this needs to say that the reference is released in the error case. Merely setting a pointer no NULL doesn't do that.

How about this:
```suggestion
.. c:function:: void PyUnicode_Append(PyObject **p_left, PyObject *right)
   Append the string *right* to the end of *p_left*.
   *p_left* must point to a :term:`strong reference` to a Unicode object;
   :c:func:`!PyUnicode_Append` releases ("steals") this reference.

   On error, set *\*p_left* to ``NULL`` and set an exception.

   On sucess, set *\*p_left* to a new strong reference to the result.
```
(There's an optimization where if the "stolen" and "new" refs are the same object, their refcounting operations can be skipped. But externally, the function behaves as above.)